### ISSUE TITLE
Line gets thrown out of scope

### ISSUE THREAD

**Issue by CD-Z** [1nVitr0/plugin-vscode-blocksort#93]
I started using this extension mainly to sort my .d.ts files. Now I just found a bug where a line gets thrown out of the block scope and causes a syntax error.
Type before sorting:
```ts
type Return<T extends any> =
  | {
      data: T;
      meta?: unknown;
      error?: undefined;
    }
  | {
      data: null;
      error: {
        status: number;
        name: string;
        message: string;
        details: object;
      };
    }

```
Type after sorting:
```ts
type Return<T extends any> =
  | {
      data: null;
      error: {
        details: object;
        message: string;
        name: string;
        status: number;
      };
    }
  | {
      data: T;
      error?: undefined;
    }
      meta?: unknown;

```

**Comment by 1nVitr0** [comment]
Might be due to the last performance update, though this seems very strange to me.

**Comment by 1nVitr0** [comment]
## TL;DR

I have found the cause for this. It will take some time to consider a general fix, that doesn't interfere with quirks of other programming languages.

### For completeness:

The detection of inner blocks works purely by indentation and completes a block only after the next line is back to the initial indentation. This results in `error?: undefined;` and `}` being grouped together into one block:

https://github.com/1nVitr0/plugin-vscode-blocksort/blob/4d3ead6ad6a4305604b461e9b1d500beb6fe33b2/src/providers/BlockSortProvider.ts#L193-L219

```
HEADER  >>> type Return<T extends any> =

BLOCK 1 >>>   | {
BLOCK 1 >>>       data: T;
BLOCK 1 >>>       meta?: unknown;
BLOCK 1 >>>       error?: undefined;
BLOCK 1 >>>     }

BLOCK 2 >>>   | {
BLOCK 2 >>>       data: null;
BLOCK 2 >>>       error: {
BLOCK 2 >>>         status: number;
BLOCK 2 >>>         name: string;
BLOCK 2 >>>         message: string;
BLOCK 2 >>>         details: object;
BLOCK 2 >>>       };
BLOCK 2 >>>     }
```

Inner Blocks for `BLOCK 1`:
```
HEADER    >>>   | {

BLOCK 1.1 >>>       data: T;

BLOCK 1.2 >>>       meta?: unknown;

BLOCK 1.3 >>>       error?: undefined;
BLOCK 1.3 >>>     }
```

Inner Blocks for `BLOCK 2`
```
HEADER    >>>   | {

BLOCK 2.1 >>>       data: null;

BLOCK 2.2 >>>       error: {
BLOCK 2.2 >>>         status: number;
BLOCK 2.2 >>>         name: string;
BLOCK 2.2 >>>         message: string;
BLOCK 2.2 >>>         details: object;
BLOCK 2.2 >>>       };
BLOCK 2.2 >>>     }
```

This isn't visible for the second block, but results in `BLOCK 1.3` being sorted in before `BLOCK 1.2` and terminating the object in an invalid way.

**Comment by 1nVitr0** [comment]
@CD-Z Should now work as expected. The main challenge was supporting a `;` after the last type in a union type while not getting confused by `;` in type and interface definitions.

This shouldn't introduce issues in other languages, but if you notice any feel free to open another issue
